src/mesh/mesh_refinement.cpp

```diff
@@ -799,11 +823,31 @@ void MeshRefinement::ProlongateSharedFieldX1(
           Real gx3p = (coarse(k+1,j,i) - ccval)/dx3p;
           Real gx3c = 0.5*(SIGN(gx3m) + SIGN(gx3p))*std::min(std::abs(gx3m),
                                                              std::abs(gx3p));
-
-          fine(fk  ,fj  ,fi) = ccval - gx2c*(x2c - fx2m) - gx3c*(x3c - fx3m);
-          fine(fk  ,fj+1,fi) = ccval + gx2c*(fx2p - x2c) - gx3c*(x3c - fx3m);
-          fine(fk+1,fj  ,fi) = ccval - gx2c*(x2c - fx2m) + gx3c*(fx3p - x3c);
-          fine(fk+1,fj+1,fi) = ccval + gx2c*(fx2p - x2c) + gx3c*(fx3p - x3c);
+          if (std::strcmp(COORDINATE_SYSTEM, "spherical_polar") == 0) {
+            Real dfx2 = fx2p - fx2m;
+            Real dfx3 = fx3p - fx3m;
+            fine(fk  ,fj  ,fi) = ccval - gx2c*dfx2*(sarea_x1_[0][1](fi)
```

Comment:
It is probably better (but not sure and can be compiler dependent) to load array quantities to local variables and reuse them as much as possible. For example,
```
Real carea = csarea_x1(i);
Real f00 = sarea_x1_[0][0](fi);
```
etc.

Reply:
In fact, all the radial face areas are similar along the same (j,k) (even when the coordinate spacing is non uniform, so we can use the values for the first i and fi to save some computational cost, although I guess this part does not dominate the cost anyway.